ProfileView/src/index.ts

```diff
@@ -171,13 +189,82 @@ export default class CjaasProfileWidget extends LitElement {
         this.requestUpdate();
       })
       .catch((err: Error) => {
-        console.log(err);
+        console.error(err);
         this.profile = undefined;
         this.showSpinner = false;
         this.requestUpdate();
       });
   }
 
+  getProfileFromTemplateId() {
+    const url = `${this.baseURL}/v1/journey/views?viewId=${this.templateId}&personId=${this.customer}`;
+
+    this.showSpinner = true;
+
+    const options: AxiosRequestConfig = {
+      url,
+      method: "POST",
+      headers: {
+        "Content-type": "application/json",
+        Authorization: "SharedAccessSignature " + this.profileReadToken,
+        "X-CACHE-MAXAGE-HOUR": 5,
+      },
+    };
+
+    axios(options)
+      .then((x) => x.data)
+      .then((x) => {
+        this.setOffProfileLongPolling(x.getUriStatusQuery);
+      })
+      .catch((err) => {
+        console.error("Unable to fetch the Profile", err);
+        this.showSpinner = false;
+      });
+  }
+
+  setOffProfileLongPolling(url: string) {
+    let intervalId = setInterval(() => {
+      axios({
+        url,
+        method: "GET",
+        headers: {
+          "Content-type": "application/json",
+          Authorization: "SharedAccessSignature " + this.profileReadToken,
+        },
+      })
+        .then((x) => x.data)
+        .then((x) => {
+          if (x.runtimeStatus === "Completed") {
+            this.showSpinner = false;
+            clearInterval(intervalId);
+            this.profile = x?.output?.ProfileView?.AttributeView.$values.map(
+              (x: any) => {
```

Comment:
Maybe nitpicky, but is there a more descriptive term than `x` that could be used here? I always find that to be a little too abstract and adds time to reading later . . . unless the case is that it truly is such an undetermined dataset as to have no better name.

Reply:
Agreed. I will amend this.